source/dub/dub.d

```diff
@@ -273,36 +277,51 @@ class Dub {
-		m_config = new DubConfig(jsonFromFile(dubFolderPath ~ "../etc/dub/settings.json", true), m_config);
+		const dubFolderSettings = (dubFolderPath ~ "../etc/dub/settings.json").toNativeString();
+		if (dubFolderSettings.exists)
+			m_config = new DubConfig(parseConfigFile!UserConfiguration(CLIArgs(dubFolderSettings)), m_config);
+
 		version (Posix) {
 			if (dubFolderPath.absolute && dubFolderPath.startsWith(NativePath("usr"))) {
-				m_config = new DubConfig(jsonFromFile(NativePath("/etc/dub/settings.json"), true), m_config);
+				const etcSettings = NativePath("/etc/dub/settings.json").toNativeString();
+				if (etcSettings.exists)
+					m_config = new DubConfig(parseConfigFile!UserConfiguration(CLIArgs(etcSettings)), m_config);
 			}
 		}
 
-		m_config = new DubConfig(jsonFromFile(m_dirs.userSettings ~ "settings.json", true), m_config);
+		const userSettings = (m_dirs.userSettings ~ "settings.json").toNativeString();
+		if (userSettings.exists)
+			m_config = new DubConfig(parseConfigFile!UserConfiguration(CLIArgs(userSettings)), m_config);
 
 		if (!root_path.empty)
-			m_config = new DubConfig(jsonFromFile(root_path ~ "dub.settings.json", true), m_config);
+		{
+			const rootSettings = (root_path ~ "dub.settings.json").toNativeString();
+			if (rootSettings.exists)
+				m_config = new DubConfig(parseConfigFile!UserConfiguration(CLIArgs(rootSettings)), m_config);
+		}
 
 		determineDefaultCompiler();
 
 		m_defaultArchitecture = m_config.defaultArchitecture;
 		m_defaultLowMemory = m_config.defaultLowMemory;
-		m_defaultEnvironments = m_config.defaultEnvironments;
-		m_defaultBuildEnvironments = m_config.defaultBuildEnvironments;
-		m_defaultRunEnvironments = m_config.defaultRunEnvironments;
-		m_defaultPreGenerateEnvironments = m_config.defaultPreGenerateEnvironments;
-		m_defaultPostGenerateEnvironments = m_config.defaultPostGenerateEnvironments;
-		m_defaultPreBuildEnvironments = m_config.defaultPreBuildEnvironments;
-		m_defaultPostBuildEnvironments = m_config.defaultPostBuildEnvironments;
-		m_defaultPreRunEnvironments = m_config.defaultPreRunEnvironments;
-		m_defaultPostRunEnvironments = m_config.defaultPostRunEnvironments;
+		m_defaultEnvironments = cast(string[string]) m_config.defaultEnvironments;
+		m_defaultBuildEnvironments = cast(string[string]) m_config.defaultBuildEnvironments;
+		m_defaultRunEnvironments = cast(string[string]) m_config.defaultRunEnvironments;
+		m_defaultPreGenerateEnvironments = cast(string[string]) m_config.defaultPreGenerateEnvironments;
+		m_defaultPostGenerateEnvironments = cast(string[string]) m_config.defaultPostGenerateEnvironments;
+		m_defaultPreBuildEnvironments = cast(string[string]) m_config.defaultPreBuildEnvironments;
+		m_defaultPostBuildEnvironments = cast(string[string]) m_config.defaultPostBuildEnvironments;
+		m_defaultPreRunEnvironments = cast(string[string]) m_config.defaultPreRunEnvironments;
+		m_defaultPostRunEnvironments = cast(string[string]) m_config.defaultPostRunEnvironments;
```

Comment:
This is a temporary hack removed in the next commit to make things compile. I probably will squash both commits.